Marlin/pins_ANET_10.h

```diff
@@ -21,84 +21,72 @@
  */
 
 /**
- * Sanguinololu board pin assignments
+ * Anet board pin assignments
  */
 
 #define BOARD_NAME "Anet"
 
-#if defined(__AVR_ATmega1284P__)
-#define LARGE_FLASH true
-#define VARIANT_STANDARD true
-#endif
-
-#define SANGUINOLOLU_V_1_2
-#include "pins_SANGUINOLOLU_11.h"
+#include "pins_SANGUINOLOLU_12.h"
 
 #undef LED_PIN
 #undef FAN_PIN
 
-#define LED_PIN -1
 #define FAN_PIN 4
 
 #if ENABLED(ULTRA_LCD) && ENABLED(NEWPANEL)
-	#undef LCD_PINS_RS
-	#undef LCD_PINS_ENABLE
-	#undef LCD_PINS_D4
-	#undef LCD_PINS_D5
-	#undef LCD_PINS_D6
-	#undef LCD_PINS_D7
-
-	#if ENABLED(ADC_KEYPAD)
-		#undef BTN_EN1
-		#undef BTN_EN2
-		#undef BTN_ENC
-
-		#define SERVO0_PIN         27 // free for BLTouch/3D-Touch
-		#define LCD_PINS_RS        28
-		#define LCD_PINS_ENABLE    29
-		#define LCD_PINS_D4        10
-		#define LCD_PINS_D5        11
-		#define LCD_PINS_D6        16
-		#define LCD_PINS_D7        17
-
-		#define BTN_EN1 			  -1
-		#define BTN_EN2 			  -1
-		#define BTN_ENC 			  -1
-
-		#define ENCODER_FEEDRATE_DEADZONE 2
-	#elif ENABLED(U8GLIB_ST7920)
-		#undef BEEPER_PIN
-
-	#if ENABLED(REPRAP_DISCOUNT_FULL_GRAPHIC_SMART_CONTROLLER)
-		// For RepRap Discount (with Anet Adapter wiring)
-		#define SERVO0_PIN        27 // free for BLTouch/3D-Touch
-		#define BEEPER_PIN        28
-		#define LCD_PINS_RS       30
-		#define	LCD_PINS_ENABLE   29
-		#define LCD_PINS_D4       17
-		#define BTN_EN1               11
-		#define BTN_EN2               10
-	#elif ENABLED(ANET_FULL_GRAPHICS_LCD)
-		#define SERVO0_PIN         29 // free for BLTouch/3D-Touch
-		#define BEEPER_PIN 17
-		#define LCD_PINS_RS        27
-		#define LCD_PINS_ENABLE    28
-		#define LCD_PINS_D4        30
-	#else
-		#error "You need to select ANET or RepRap Version"
-	#endif
-
-		#define ST7920_DELAY_1 DELAY_0_NOP
-		#define ST7920_DELAY_2 DELAY_1_NOP
-		#define ST7920_DELAY_3 DELAY_2_NOP
-
-		#ifndef ENCODER_STEPS_PER_MENU_ITEM
-			#define ENCODER_STEPS_PER_MENU_ITEM 1
-		#endif
-		#ifndef ENCODER_PULSES_PER_STEP
-			#define ENCODER_PULSES_PER_STEP 4
-		#endif
-	#endif
+  #undef LCD_PINS_RS
+  #undef LCD_PINS_ENABLE
+  #undef LCD_PINS_D4
+  #undef LCD_PINS_D5
+  #undef LCD_PINS_D6
+  #undef LCD_PINS_D7
+
+  #if ENABLED(ADC_KEYPAD)
+    #undef BTN_EN1
+    #undef BTN_EN2
+    #undef BTN_ENC
+
+    #define SERVO0_PIN         27 // free for BLTouch/3D-Touch
+    #define LCD_PINS_RS        28
+    #define LCD_PINS_ENABLE    29
+    #define LCD_PINS_D4        10
+    #define LCD_PINS_D5        11
+    #define LCD_PINS_D6        16
+    #define LCD_PINS_D7        17
+
+    #define BTN_EN1         -1
+    #define BTN_EN2         -1
+    #define BTN_ENC         -1
+
```

Comment:
I added the pin to the "base" branch and changed the other code. pin is now always there if adc_keypad is used.